crates/core/tedge/src/cli/init.rs

```diff
@@ -1,26 +1,102 @@
+use anyhow::bail;
 use anyhow::Context;
+use clap::Subcommand;
+use std::os::unix::fs::MetadataExt;
+use std::path::Path;
 use tedge_utils::file::create_directory;
 use tedge_utils::file::PermissionEntry;
 
 #[derive(Debug)]
 pub struct TEdgeInitCmd {
     user: String,
     group: String,
+    relative_links: bool,
     context: BuildContext,
 }
 
 impl TEdgeInitCmd {
-    pub fn new(user: String, group: String, context: BuildContext) -> Self {
+    pub fn new(user: String, group: String, relative_links: bool, context: BuildContext) -> Self {
         Self {
             user,
             group,
+            relative_links,
             context,
         }
     }
 
     fn initialize_tedge(&self) -> anyhow::Result<()> {
+        let executable_name =
+            std::env::current_exe().context("retrieving the current executable name")?;
+        let stat = std::fs::metadata(&executable_name).with_context(|| {
+            format!(
+                "reading metadata for the current executable ({})",
+                executable_name.display()
+            )
+        })?;
+        let Some(executable_dir) = executable_name.parent() else {
+            bail!(
+                "current executable ({}) does not have a parent directory",
+                executable_name.display()
+            )
+        };
+
+        let component_subcommands: Vec<String> =
+            Component::augment_subcommands(clap::Command::new("tedge"))
+                .get_subcommands()
+                .map(|c| c.get_name().to_owned())
+                .chain(["tedge-apt-plugin".to_owned()])
+                .collect();
+
+        for component in &component_subcommands {
+            let link = executable_dir.join(component);
+            match std::fs::symlink_metadata(&link) {
+                Err(e) if e.kind() != std::io::ErrorKind::NotFound => bail!(
+                    "couldn't read metadata for {}. do you need to run with sudo?",
+                    link.display()
+                ),
+                meta => {
+                    let file_exists = meta.is_ok();
+                    if file_exists {
+                        nix::unistd::unlink(&link).with_context(|| {
+                            format!("removing old version of {component} at {}", link.display())
+                        })?;
+                    }
+
+                    let tedge = if self.relative_links {
+                        // TODO this could lead to broken links
+                        // TODO is this feature even that useful?
+                        Path::new("./tedge")
+                    } else {
+                        &*executable_name
+                    };
+                    std::os::unix::fs::symlink(tedge, &link).with_context(|| {
+                        format!("creating symlink for {component} to {}", tedge.display())
+                    })?;
+
+                    let res = std::process::Command::new("chown")
+                        .arg("-h")
```

Comment:
nit: Surprised that in chown, `-h` is not `--help`. For readability I think it's be better to use long flags, so `--no-dereference` in this case.